base/shell/explorer/startup.cpp

```diff
@@ -478,11 +478,16 @@ INT ProcessStartupItems(VOID)
      */
     res = TRUE;
 
-    if (res && bNormalBoot)
-        ProcessRunOnceEx(HKEY_LOCAL_MACHINE);
+    if (bRunOnce)
+    {
+        if (res && bNormalBoot)
+            ProcessRunOnceEx(HKEY_LOCAL_MACHINE);
+
+        if (res && (SHRestricted(REST_NOLOCALMACHINERUNONCE) == 0))
```

Comment:
Should this be: `res && bNormalBoot & (SHRestricted(...) == 0)`, as this seems to be implied by all the other checks in the function?

Reply:
No, MSDN documents that if the value (Name? Command?) begins with a *, it should be executed in safe mode (there is a TODO comment here in this function).

Reply:
> [By default, these keys are ignored when the computer is started in Safe Mode. The value name of RunOnce keys can be prefixed with an asterisk (*) to force the program to run even in Safe Mode.](https://learn.microsoft.com/en-us/windows/win32/setupapi/run-and-runonce-registry-keys)

I assume they mean value name but would have to be tested in another PR.

Reply:
Thanks for the info